test/test_Data.py

```diff
@@ -39,11 +41,12 @@ def insert_data_from_csv(self: TestBase, in_csv_file, rec_count=-1):
     if self.stats:
         print("\n")
 
-    loader = ParallelLoader(self.db)
+    loader = ParallelLoader(self.db,
+                            response_handler=response_handler)
     loader.ingest(data, batchsize=self.batchsize,
                   numthreads=self.numthreads,
                   stats=self.stats)
-    self.assertEqual(loader.error_counter, 0)
+    assert loader.error_counter == 0
```

Comment:
Ah so this might be a longish explanation.
The way we have used parameterization is based on the pytest. This is actually quite different from the way has such tests. I myself discovered the details while working on this PR. If interested, the docs are here : https://pytest.org/en/7.1.x/how-to/unittest.html

So, to discover pytest type parameterized tests, I could not use a class derived from unittest.TestCase. Hence these tests do not derive from TestBase. This implies that this class does not have self.assert* methods. But the way pytest renders the failures is IMO more useful, and intuitive.

Let me know if you need more information on the same.